src/lib/4337.js

```diff
@@ -0,0 +1,249 @@
+import { ethers } from "ethers";
+import tokenEntryPointContractAbi from "smartcontracts/build/contracts/tokenEntryPoint/TokenEntryPoint.abi";
+import accountFactoryContractAbi from "smartcontracts/build/contracts/accfactory/AccountFactory.abi";
+import accountContractAbi from "smartcontracts/build/contracts/account/Account.abi";
+import tokenContractAbi from "smartcontracts/build/contracts/erc20/ERC20.abi";
+
+const accountFactoryInterface = new ethers.utils.Interface(
+  accountFactoryContractAbi
+);
+const accountInterface = new ethers.utils.Interface(accountContractAbi);
+const erc20Token = new ethers.utils.Interface(tokenContractAbi);
+
+const adjustForDecimals = (amount, decimals) => {
+  return amount * 10 ** decimals;
+};
+
+const transferCallData = (tokenAddress, receiver, amount) =>
+  accountInterface.encodeFunctionData("execute", [
+    tokenAddress,
+    ethers.constants.Zero,
```

Comment:
what does this second argument refer to?

Reply:
The second argument to account creation on a factory is a nonce. It's usually set to 0.